test/browser/push.test.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+define(['ably', 'shared_helper', 'chai', 'push'], function (Ably, helper, chai, PushPlugin) {
+  const expect = chai.expect;
+  const whenPromiseSettles = helper.whenPromiseSettles;
+  const swUrl = '/push_sw.js';
+  let rest;
+
+  const persistKeys = {
+    deviceId: 'ably.push.deviceId',
+    deviceSecret: 'ably.push.deviceSecret',
+    deviceIdentityToken: 'ably.push.deviceIdentityToken',
+    pushRecipient: 'ably.push.pushRecipient',
+    activationState: 'ably.push.activationState',
+  };
+
+  const messageChannel = new MessageChannel();
+
+  /**
+   * These tests don't work in CI for various reasons (below) but should work when running locally via `npm run test:webserver`, provided
+   * that you have enabled notification permissions for the origin serving the test ui.
+   *
+   * chromium, firefox - don't support push notifications in incognito
+   * webkit - doesn't have a way to launch programatically with notification permissions granted
+   */
+  if (Notification.permission === 'granted') {
+    describe('browser/push', function () {
+      this.timeout(60 * 1000);
+
+      before(function (done) {
+        helper.setupApp(function () {
+          rest = helper.AblyRest({
+            pushServiceWorkerUrl: swUrl,
+            plugins: { Push: PushPlugin },
+          });
+          done();
+        });
+      });
+
+      beforeEach(async function () {
+        Object.values(persistKeys).forEach((key) => {
+          Ably.Realtime.Platform.Config.push.storage.remove(key);
+        });
+
+        const worker = await navigator.serviceWorker.getRegistration(swUrl);
+
+        if (worker) {
+          await worker.unregister();
+        }
+      });
+
+      afterEach(async function () {
+        await rest.push.deactivate();
+      });
+
+      it('push_activation_succeeds', async function () {
```

Comment:
yeah i opted not to tag these ones since they test the same spec points as existing tests in `test/rest/push.test.js`, but using the web push transport rather than push channel transport, i'll add tags to them now though